src/eradiate/scenes/measure/_multi_distant.py

```diff
@@ -126,15 +143,21 @@ def viewing_angles(self) -> pint.Quantity:
             dimension is ordered as (zenith, azimuth).
         """
         angle_units = ucc.get("angle")
-        angles = direction_to_angles(-self.directions).to(angle_units)
+        angles = frame.direction_to_angles(-self.directions).m_as(ureg.rad)
 
         # Snap zero values to avoid close-to-360° azimuths
         angles[:, 1] = np.where(np.isclose(angles[:, 1], 0.0), 0.0, angles[:, 1])
```

Comment:
Let us assume that we are computing the angles corresponding to a set of directions in the 0° hemisphere plane. The computed azimuth values will be close to 0, but usually not exactly equal because of floating point arithmetic precision. This is theoretically harmless but in practice, the normalisation to the [0, 2π] interval doesn't play well with it: values of -ε are then normalised to 360-ε, and values of +ε are left unchanged. You end up with angles wiggling between ε and 360-ε (possibly rounded to 360 by machine precision), and this is not nice e.g.for plotting.